lightning/src/ln/peer_handler.rs

```diff
@@ -1323,7 +1346,8 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 
 		// Need an Init as first message
 		if let wire::Message::Init(msg) = message {
-			if msg.features.requires_unknown_bits() {
+			let additional_features = msg.features.clone() - self.init_features(&their_node_id);
```

Comment:
Mostly to be able to reuse `requires_unknown_bits`.

Reply:
Yeah, it seems `requires_unknown_bits` is obsolete given it uses `KNOWN_FEATURE_MASK`. A set of handlers may exclude a known feature. If the peer requires the feature, the method would return false instead of true. We really want to use a mask specific to the features supported by the handlers.

Reply:
Ok, I think this is implemented correctly now. PTAL. I can remove the set difference code when I squash the fixups.